### DISCUSSION TITLE
Is it expected that send() won't work in transitions to final states?

### DISCUSSION THREAD

**Discussion by FokkeZB** [statelyai/xstate#1951] [Q&A]
We're using a global transition to track events. Anywhere we need to track an event, we use `send({ type: 'TRACK_EVENT', data })` or even `send((ctx, evt) => ({ type: 'TRACK_EVENT', data: { ctx.something, evt.something }))`.

I noticed that when this is done in a transition that targets a final state, these actions won't be executed.

I'd expect any actions to run _before_ advancing to the target state.

See https://codesandbox.io/s/festive-napier-ps5yp?file=/src/index.js for a full example.

Here's the machine:

```
const machine = createMachine({
  id: "machine",
  initial: "start",
  states: {
    start: {
      on: {
        END: {
          actions: [
            () => {
              // Always logs
              console.info("END.actions[0]");
            },
            send({ type: "SELF_TRANSITION", data: "END.actions[1]" })
          ],
          target: "end"
        }
      }
    },
    end: {
      // type: "final"
    }
  },
  on: {
    SELF_TRANSITION: {
      actions: (_, { data }) => {
        // Won't log when end.type === 'final'
        console.info("SELF_TRANSITION.actions", data);
      }
    }
  }
});
```

**Comment by davidkpiano** [comment]
When you reach a final state on the root of the machine, the machine is "done" - no other transitions can follow, so no other actions can be executed.

**Reply by Andarist** [reply]
This should actually be executed as it's part of the [`executeTransitionContent`](https://www.w3.org/TR/scxml/#executeTransitionContent) procedure that is called by the [`microstep`](https://www.w3.org/TR/scxml/#microstepProcedure) and at that point, the target state's type and the resulting configuration is not taken into account - this only happens at a latter step of `microstep`, it's [`enterStates`](https://www.w3.org/TR/scxml/#enterStates) that sets `running = false`.

I would classify this is a bug and report it as an issue.

**Reply by FokkeZB** [reply]
Haha, ok! Reported at https://github.com/davidkpiano/xstate/issues/1953
